src/System.IO.Ports/src/System/IO/Ports/SerialStream.Unix.cs

```diff
@@ -0,0 +1,511 @@
+            if (array.Length - offset < count)
+                throw new ArgumentException(SR.Argument_InvalidOffLen);
+            Debug.Assert(timeout == SerialPort.InfiniteTimeout || timeout >= 0, "Serial Stream Write - write timeout is " + timeout);
+
+            // check for open handle, though the port is always supposed to be open
+            if (_handle == null) throw new IOException(); //InternalResources.FileNotOpen();
+
+            int numBytes;
+            if (_isAsync)
+            {
+                throw new NotSupportedException(SR.PlatformNotSupported_IOPorts);
+            }
+            else
+            {
+                fixed (byte* bufPtr = array)
+                {
+                    numBytes = Interop.Sys.Write(_handle, bufPtr, count);
+                }
+
+                if (numBytes == -1)
+                {
+                    throw new IOException();
+                }
+            }
+
+            if (numBytes == 0)
+                throw new TimeoutException(SR.Write_timed_out);
+
+        }
+
+        internal SafeFileHandle OpenPort(string portName)
+        {
+            // TBD O_NOCTTY | O_NDELAY ???
+            SafeFileHandle handle = Interop.Sys.Open(portName, Interop.Sys.OpenFlags.O_RDWR | Interop.Sys.OpenFlags.O_CLOEXEC, 0);
+
+            return handle;
+        }
+
+       // this method is used by SerialPort upon SerialStream's creation
+        internal SerialStream(string portName, int baudRate, Parity parity, int dataBits, StopBits stopBits, int readTimeout, int writeTimeout, Handshake handshake,
+            bool dtrEnable, bool rtsEnable, bool discardNull, byte parityReplace)
+        {
+
+            if (portName == null)
+            {
+                throw new ArgumentException(SR.Arg_InvalidSerialPort, nameof(portName));
+            }
+
+            // Error checking done in SerialPort.
+
+            SafeFileHandle tempHandle = OpenPort(portName);
+
+            if (tempHandle.IsInvalid)
+            {
+                throw new ArgumentException(SR.Arg_InvalidSerialPort, nameof(portName));
+            }
+
+            try
+            {
+                _handle = tempHandle;
+                // set properties of the stream that exist as members in SerialStream
+                _portName = portName;
+                _handshake = handshake;
+                _parity = parity;
+                _readTimeout = readTimeout;
+                _writeTimeout = writeTimeout;
+                _baudRate = baudRate;
+                _stopBits = stopBits;
+                _dataBits = dataBits;
+                _parity = parity;
+
+                selectList = new ArrayList();
+                if (Interop.Sys.TerminalReset(_handle, _baudRate, _dataBits, _stopBits, _parity, _handshake) != 0)
+                {
+                    throw new ArgumentException();
+                }
+
+                DtrEnable = dtrEnable;
+                // query and cache the initial RtsEnable value
```

Comment:
The comment does not match the actual code `_rtsEnable` is written to but not read anywhere.